src/rerank/rerank_utils.cpp

```diff
@@ -187,4 +187,104 @@ std::optional<bool> RerankHandler::getReturnDocuments() const { return request.r
 std::optional<std::vector<std::string>> RerankHandler::getRankFields() const { return request.rankFields; }
 std::optional<int> RerankHandler::getMaxChunksPerDoc() const { return request.maxChunksPerDoc; }
 
+absl::Status chunkDocuments(
+    const ov::Tensor& in_input_ids, const ov::Tensor& in_attention_mask,
+    ov::Tensor& out_input_ids, ov::Tensor& out_attention_mask,
+    std::vector<size_t>& chunk_mapping, size_t max_tokens_per_chunk,
+    int64_t pad_token) {
+
```

Comment:
comment would be useful with explanations what this method does

Reply:
doesnt unit test case explain it? with example input and output
if not, i can also describe it here

Reply:
added